fix(html-validator): ensure compatibility

### Environment

- Operating System: Linux
- Node Version:     v22.14.0
- Nuxt Version:     3.16.1
- CLI Version:      3.24.0
- Nitro Version:    2.11.8
- Package Manager:  pnpm@9.15.9
- Builder:          -
- User Config:      modules, devtools, future, compatibilityDate, content
- Runtime Modules:  @nuxt/content@3.4.0, @nuxtjs/html-validator@2.1.0
- Build Modules:    -

### Version

v3.4.0

### Reproduction

https://stackblitz.com/edit/github-l2iaunq8?file=nuxt.config.ts

Make sure to download the project and run it with a modern Node version for SQLite support!

Then use `pnpm build` to get the logs as shown below.

### Description

When having [`@nuxtjs/html-validator`](https://github.com/nuxt-modules/html-validator), the validator tries to parse the SQLite file's doctype (`/__nuxt_content/content/sql_dump`) as that route is part of prerendering. Should the database file be part of prerendering? It seems to be the only prerendered page, even in a large project of mine with many "actual" pages, which makes me think it shouldn't.

### Additional context

```
 ERROR  HTML validation errors found for /__nuxt_content/content/sql_dump                                                                                                       12:09:38 AM

inline
  1:1  error  Document is missing doctype  missing-doctype

✖ 1 problem (1 error, 0 warnings)

More information:
  https://html-validate.org/rules/missing-doctype.html
```

### Logs

```shell-script

```

--- a/src/runtime/presets/cloudflare-pages/database-handler.ts
+++ b/src/runtime/presets/cloudflare-pages/database-handler.ts
@@ -1,8 +1,9 @@
-import { eventHandler, getRouterParam } from 'h3'
+import { eventHandler, getRouterParam, setHeader } from 'h3'
 import { useStorage } from 'nitropack/runtime'
 
 export default eventHandler(async (event) => {
   const collection = getRouterParam(event, 'collection')!
+  setHeader(event, 'Content-Type', 'text/plain')
 
   const ASSETS = event?.context?.cloudflare?.env.ASSETS || process.env.ASSETS
   if (ASSETS) {

--- a/src/runtime/presets/node/database-handler.ts
+++ b/src/runtime/presets/node/database-handler.ts
@@ -1,8 +1,9 @@
-import { eventHandler, getRouterParam } from 'h3'
+import { eventHandler, getRouterParam, setHeader } from 'h3'
 import { useStorage } from 'nitropack/runtime'
 
 export default eventHandler(async (event) => {
   const collection = getRouterParam(event, 'collection')!
+  setHeader(event, 'Content-Type', 'text/plain')
 
   const data = await useStorage().getItem(`build:content:database.compressed.mjs`) || ''
   if (data) {

